website/docs/d/config_map.html.markdown

```diff
@@ -0,0 +1,50 @@
+---
+layout: "kubernetes"
+page_title: "Kubernetes: kubernetes_config map"
+sidebar_current: "docs-kubernetes-resource-config map"
+description: |-
+  This data source reads configuration data from a config map.
+---
+
+# kubernetes_config map
+
+Config Maps are key-value pairs containing configuration data. The Config Map data source provides a mechanism for extracting these key-value pairs.
+
+~> **Note:** All arguments including the config map data will be stored in the raw state as plain-text. [Read more about sensitive data in state](/docs/state/sensitive-data.html).
```

Comment:
I think this note only applies to the ConfigMap resource, not its data source.

Reply:
Good point. I actually copied all this from the Secret data source, so I should probably also fix that in the Secret data source.

Reply:
Oh actually, look at this: the data source stores the data in the tfstate.

```
    {
      "mode": "data",
      "type": "kubernetes_config_map",
      "name": "example",
      "provider": "provider.kubernetes",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "binary_data": {
              "key1": "VVRGLTggZGVjb2RlciBjYXBhYmlsaXR5IGFuZCBzdHJlc3MgdGV
...
```

So I think it makes sense to leave this warning here, so users are aware that any data they read in via the data source is going to be stored in plaintext. Even though I know it's bad practice to put sensitive data into a config map, I've seen it in production at another place I worked. What do you think?

Reply:
Yep you're right, makes sense to leave it in.